gui-whelktool/src/main/java/whelk/gui/RunPanel.java

```diff
@@ -0,0 +1,142 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class RunPanel extends WizardCard implements ActionListener {
+
+    private Wizard wizard;
+
+    private JTextArea stdErrArea;
+    private JScrollPane stdErrScroll;
+    private JPasswordField passwordField;
+    private JButton startButton;
+
+    private PortableScript scriptToRun;
+
+    private AtomicBoolean scriptIsDone = new AtomicBoolean(false);
+
+    public RunPanel(Wizard wizard)
+    {
+        super(wizard);
+
+        stdErrArea = new JTextArea();
+        stdErrArea.setFont(new Font("monospaced", Font.PLAIN, 12));
+        stdErrScroll = new JScrollPane(stdErrArea);
+        stdErrScroll.setPreferredSize(new Dimension(300, 300));
+        stdErrArea.setEditable(false);
+
+        Box stdErrBox = Box.createVerticalBox();
+        stdErrBox.add(new JLabel("Körningslogg:"));
+        stdErrBox.add(stdErrScroll);
+        this.add(stdErrBox);
+
+        Box vbox = Box.createVerticalBox();
+
+        vbox.add(new JLabel("Databasens lösenord:"));
+        passwordField = new JPasswordField();
+        vbox.add(passwordField);
+
+        vbox.add(Box.createVerticalStrut(10));
+        startButton = new JButton("Starta");
+        startButton.setActionCommand("start");
+        startButton.addActionListener(this);
+        vbox.add(startButton);
+
+        this.add(vbox);
+    }
+
+    @Override
+    void onShow(Object parameterFromPreviousCard)
+    {
+        scriptToRun = (PortableScript) parameterFromPreviousCard;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent actionEvent)
+    {
+        if (actionEvent.getActionCommand().equals("start"))
+        {
+            startButton.setEnabled(false);
+            passwordField.setEnabled(false);
+
+            ByteArrayOutputStream stdErrStream = new ByteArrayOutputStream();
+            System.setErr(new PrintStream(stdErrStream));
+
+            disableNext();
+            disableBack();
+            disableCancel();
+
+            new Thread(new Runnable() {
+                public void run()
+                {
+                    try
+                    {
+                        String secretProperties = "sqlUrl = " +
+                                System.getProperty("secretSqlUrl").
+                                        replace("_XL_PASSWORD_", new String(passwordField.getPassword())) + "\n" +
+                                "baseUri = " +
+                                System.getProperty("secretBaseUri") + "\n" +
+                                "elasticHost = " +
+                                System.getProperty("secretElasticHost") + "\n" +
+                                "elasticCluster = " +
+                                System.getProperty("secretElasticCluster") + "\n" +
+                                "elasticIndex = " +
+                                System.getProperty("secretElasticIndex") + "\n";
```

Comment:
Här skulle det vara bra om man även kunde köra med en vanlig `xl.secret.properties` istället. 
Om man som jag råkar ha annat format på parametrarna.

Reply:
Väsentligen argumenterar du för att byta prio ordning ifrån:
1. Environment parameter pointing to a file, like so -Dxl.secret.properties=./secret.properties
2. User entered parameters, registered through a previous call to setUserEnteredProperties()
3. Properties files on the classpath (for example in src/main/resources/secret.properties)

Och byta plats på 2 och 3, är det korrekt uppfattat?

Kanske möjligen disable password fältet om man inte fått in övriga delar av secret från miljön?

(du kan redan köra med en secret.properties)

Reply:
Funkar inte eftersom man får en NPE på rad 89 om man inte satt den systempropertyn

Reply:
dvs `java -Dxl.secret.properties=/home/olov/secret.properties-local -jar build/libs/gui-whelktool.jar` funkar inte.

Reply:
Men jag pushade en ny version som fixade det för nån timma sen :)